partiql-ast/src/main/java/org/partiql/ast/AttributeConstraint.java

```diff
@@ -0,0 +1,105 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.expr.Expr;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * TODO docs, equals, hashcode
+ */
+public abstract class AttributeConstraint extends AstNode {
+
+    // NULL & NOT NULL
```

Comment:
Could we actually just split this into two classes? `Null` and `NotNull`

Reply:
I think the boolean makes the constant folding a bit easier. 

Consider: 
```
foo INT2 NOT NULL NULL NULL NOT NULL
```

you would just grab all AttributeConstraint with Type `NULL` and folding as you wish.

Reply:
I see. I guess I'm fine leaving it as one class since we follow a similar pattern for other negatable exprs like `BETWEEN` and `LIKE`.

Out of curiosity, for your above example what would be the constant-folded equivalent?

Reply:
Discussed offline. 

The constant-folded equivalent maybe latest wins or invalid semantics.